File: c++/data_src_dependent/bst_eq/bst_eq.cpp

#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"
#include "book.h"
#include "BSTElement.h"
#include "BST.h"


using namespace bridges;

int max_quakes = 1000;

int main(int argc, char **argv) {
	string hilite_color = "orange",
		   def_color = "green",
		   end_color = "red";

#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
#endif
	// read the earth quake  data and build the BST
	bridges.set_title("Recent Earthquakes (USGIS Data)");

	DataSource ds (&bridges);
	vector<EarthquakeUSGS> eq_list = ds.getEarthquakeUSGSData(max_quakes);

	BST<float, EarthquakeUSGS> bst;

	for (int k = 0; k < max_quakes; k++) {
		EarthquakeUSGS eq = eq_list[k];
		bst.insert(eq.getMagnitude(), eq);
	}

	// visualize the binary search tree
	bridges.setDataStructure(bst.getRoot());
	bridges.visualize();

	return 0;
}

File: python/data_src_dependent/bst_eq/bst_eq.py
from bridges.bridges import *
from bridges.data_src_dependent.data_source import *
from data_src_dependent.bst_eq.BST import *
import sys

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials

    bridges = Bridges(int(args[0]), args[1], args[2])

    if len(args) > 3:
        bridges.connector.set_server(args[3])

    bridges.set_title("Recent Earthquakes(USGIS Data)")

    eqlist = get_earthquake_usgs_data(100)
    bst = BST()
    for i in range(len(eqlist)):
        if eqlist[i].get_magnitude() > 3:
            bst.insert(eqlist[i].get_magnitude(), eqlist[i])
            print(bst.node_count)

    bridges.set_data_structure(bst.get_tree_root())
    bridges.visualize()


if __name__ == '__main__':
    main()